Support Java 8

When I run 
mvn clean package -DskipTests

adam-core does not compile with following stack trace:
```
[ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:3.1.5:compile (scala-compile-first) on project adam-core: Execution scala-compile-first of goal net.alchim31.maven:scala-maven-plugin:3.1.5:compile failed. CompileFailed -

[ERROR] An annotation processor threw an uncaught exception.
[ERROR] Consult the following stack trace for details.
[ERROR] java.lang.NoClassDefFoundError: com/sun/tools/javac/main/OptionName
[ERROR] 	at com.google.java.contract.core.apt.AnnotationProcessor.setupPaths(AnnotationProcessor.java:225)
[ERROR] 	at com.google.java.contract.core.apt.AnnotationProcessor.init(AnnotationProcessor.java:148)
[ERROR] 	at com.sun.tools.javac.processing.JavacProcessingEnvironment$ProcessorState.<init>(JavacProcessingEnvironment.java:500)
[ERROR] 	at com.sun.tools.javac.processing.JavacProcessingEnvironment$DiscoveredProcessors$ProcessorStateIterator.next(JavacProcessingEnvironment.java:597)
[ERROR] 	at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:690)
[ERROR] 	at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)
[ERROR] 	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)
[ERROR] 	at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)
[ERROR] 	at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)
[ERROR] 	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)
[ERROR] 	at com.sun.tools.javac.main.Main.compile(Main.java:523)
[ERROR] 	at com.sun.tools.javac.main.Main.compile(Main.java:381)
[ERROR] 	at com.sun.tools.javac.main.Main.compile(Main.java:370)
[ERROR] 	at com.sun.tools.javac.main.Main.compile(Main.java:361)
[ERROR] 	at com.sun.tools.javac.Main.compile(Main.java:56)
[ERROR] 	at com.sun.tools.javac.Main.main(Main.java:42)
[ERROR] Caused by: java.lang.ClassNotFoundException: com.sun.tools.javac.main.OptionName
[ERROR] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:372)
[ERROR] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
[ERROR] 	at java.security.AccessController.doPrivileged(Native Method)
[ERROR] 	at java.net.URLClassLoader.findClass(URLClassLoader.java:360)
[ERROR] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
[ERROR] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
[ERROR] 	... 16 more
```

P.S. BTW, why don't you move to sbt? SBT is much better for scala projects than maven and provides a lot of useful features (like incremental compilation, source-code generation, cross-scala build, nice task semantic, etc.) out of the box.

IIRC, the OptionName error is related to which version of Java you're
running. Try switching to 7 if you're using 8, or vice versa.

On Wed Jan 07 2015 at 9:56:18 AM Anton Kulaga <notifications@github.com>
wrote:

> When I run
> mvn clean package -DskipTests
>
> adam-core does not compile with following stack trace:
>
> [ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:3.1.5:compile (scala-compile-first) on project adam-core: Execution scala-compile-first of goal net.alchim31.maven:scala-maven-plugin:3.1.5:compile failed. CompileFailed -
>
> [ERROR] An annotation processor threw an uncaught exception.
> [ERROR] Consult the following stack trace for details.
> [ERROR] java.lang.NoClassDefFoundError: com/sun/tools/javac/main/OptionName
> [ERROR]     at com.google.java.contract.core.apt.AnnotationProcessor.setupPaths(AnnotationProcessor.java:225)
> [ERROR]     at com.google.java.contract.core.apt.AnnotationProcessor.init(AnnotationProcessor.java:148)
> [ERROR]     at com.sun.tools.javac.processing.JavacProcessingEnvironment$ProcessorState.<init>(JavacProcessingEnvironment.java:500)
> [ERROR]     at com.sun.tools.javac.processing.JavacProcessingEnvironment$DiscoveredProcessors$ProcessorStateIterator.next(JavacProcessingEnvironment.java:597)
> [ERROR]     at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:690)
> [ERROR]     at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)
> [ERROR]     at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)
> [ERROR]     at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)
> [ERROR]     at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)
> [ERROR]     at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)
> [ERROR]     at com.sun.tools.javac.main.Main.compile(Main.java:523)
> [ERROR]     at com.sun.tools.javac.main.Main.compile(Main.java:381)
> [ERROR]     at com.sun.tools.javac.main.Main.compile(Main.java:370)
> [ERROR]     at com.sun.tools.javac.main.Main.compile(Main.java:361)
> [ERROR]     at com.sun.tools.javac.Main.compile(Main.java:56)
> [ERROR]     at com.sun.tools.javac.Main.main(Main.java:42)
> [ERROR] Caused by: java.lang.ClassNotFoundException: com.sun.tools.javac.main.OptionName
> [ERROR]     at java.net.URLClassLoader$1.run(URLClassLoader.java:372)
> [ERROR]     at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
> [ERROR]     at java.security.AccessController.doPrivileged(Native Method)
> [ERROR]     at java.net.URLClassLoader.findClass(URLClassLoader.java:360)
> [ERROR]     at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
> [ERROR]     at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
> [ERROR]     ... 16 more
>
> P.S. BTW, why don't you move to sbt? SBT is much better for scala projects
> than maven and provides a lot of useful features (like incremental
> compilation, source-code generation, cross-scala build, nice task semantic,
> etc.) out of the box.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bigdatagenomics/adam/issues/540>.
>

cf. #198 <https://github.com/bigdatagenomics/adam/issues/198>

On Wed Jan 07 2015 at 9:56:18 AM Anton Kulaga <notifications@github.com>
wrote:

> When I run
> mvn clean package -DskipTests
>
> adam-core does not compile with following stack trace:
>
> [ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:3.1.5:compile (scala-compile-first) on project adam-core: Execution scala-compile-first of goal net.alchim31.maven:scala-maven-plugin:3.1.5:compile failed. CompileFailed -
>
> [ERROR] An annotation processor threw an uncaught exception.
> [ERROR] Consult the following stack trace for details.
> [ERROR] java.lang.NoClassDefFoundError: com/sun/tools/javac/main/OptionName
> [ERROR]     at com.google.java.contract.core.apt.AnnotationProcessor.setupPaths(AnnotationProcessor.java:225)
> [ERROR]     at com.google.java.contract.core.apt.AnnotationProcessor.init(AnnotationProcessor.java:148)
> [ERROR]     at com.sun.tools.javac.processing.JavacProcessingEnvironment$ProcessorState.<init>(JavacProcessingEnvironment.java:500)
> [ERROR]     at com.sun.tools.javac.processing.JavacProcessingEnvironment$DiscoveredProcessors$ProcessorStateIterator.next(JavacProcessingEnvironment.java:597)
> [ERROR]     at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:690)
> [ERROR]     at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)
> [ERROR]     at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)
> [ERROR]     at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)
> [ERROR]     at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)
> [ERROR]     at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)
> [ERROR]     at com.sun.tools.javac.main.Main.compile(Main.java:523)
> [ERROR]     at com.sun.tools.javac.main.Main.compile(Main.java:381)
> [ERROR]     at com.sun.tools.javac.main.Main.compile(Main.java:370)
> [ERROR]     at com.sun.tools.javac.main.Main.compile(Main.java:361)
> [ERROR]     at com.sun.tools.javac.Main.compile(Main.java:56)
> [ERROR]     at com.sun.tools.javac.Main.main(Main.java:42)
> [ERROR] Caused by: java.lang.ClassNotFoundException: com.sun.tools.javac.main.OptionName
> [ERROR]     at java.net.URLClassLoader$1.run(URLClassLoader.java:372)
> [ERROR]     at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
> [ERROR]     at java.security.AccessController.doPrivileged(Native Method)
> [ERROR]     at java.net.URLClassLoader.findClass(URLClassLoader.java:360)
> [ERROR]     at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
> [ERROR]     at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
> [ERROR]     ... 16 more
>
> P.S. BTW, why don't you move to sbt? SBT is much better for scala projects
> than maven and provides a lot of useful features (like incremental
> compilation, source-code generation, cross-scala build, nice task semantic,
> etc.) out of the box.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bigdatagenomics/adam/issues/540>.
>

I'm having a hard time compiling with the latest java 7. Can't compile with 6, as it's not supported, and can't with 8 for obvious reasons.
```~/Workbench/git/spark/adam(branch:master*) » mvn clean package -DskipTests                     
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO]
[INFO] ADAM
[INFO] ADAM: Core
[INFO] ADAM: APIs for Java
[INFO] ADAM: CLI
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building ADAM 0.16.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ adam-parent ---
[INFO] Deleting /Users/mlyons/Workbench/git/spark/adam/target
[INFO]
[INFO] --- maven-enforcer-plugin:1.0:enforce (enforce-versions) @ adam-parent ---
[INFO]
[INFO] --- maven-enforcer-plugin:1.0:enforce (enforce-maven) @ adam-parent ---
[INFO]
[INFO] --- scalariform-maven-plugin:0.1.4:format (default-cli) @ adam-parent ---
[INFO] Modified 0 of 205 .scala files
[INFO]
[INFO] --- scala-maven-plugin:3.1.5:compile (scala-compile-first) @ adam-parent ---
[INFO] No sources to compile
[INFO]
[INFO] --- scala-maven-plugin:3.1.5:testCompile (scala-test-compile-first) @ adam-parent ---
[INFO] No sources to compile
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Building ADAM: Core 0.16.1-SNAPSHOT
[INFO] ------------------------------------------------------------------------
[INFO]
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ adam-core ---
[INFO] Deleting /Users/mlyons/Workbench/git/spark/adam/adam-core/target
[INFO]
[INFO] --- maven-enforcer-plugin:1.0:enforce (enforce-versions) @ adam-core ---
[INFO]
[INFO] --- maven-enforcer-plugin:1.0:enforce (enforce-maven) @ adam-core ---
[INFO]
[INFO] --- git-commit-id-plugin:2.1.9:revision (default) @ adam-core ---
[info] dotGitDirectory /Users/mlyons/Workbench/git/spark/adam/.git
[info] git.build.user.name mike_lyons
[info] git.build.user.email mike.lyons@invitae.com
[info] git.branch master
[info] Tag refs [ [Ref[refs/tags/adam-parent-0.10.0=313086f04601dc782692d060e6f708dd09ff0a18], Ref[refs/tags/adam-parent-0.11.0=0699a48a62f72eaae134dbc7e99cf68400ef46fd], Ref[refs/tags/adam-parent-0.12.0=d923656d2ed75c86530308314a55c5640f9ea3e8], Ref[refs/tags/adam-parent-0.12.1=c75c4dc7b49bd66b0953e2413b7df6802df13c32], Ref[refs/tags/adam-parent-0.13.0=50557f60e03ebaaa07316578eca6864f5522ac18], Ref[refs/tags/adam-parent-0.14.0=5ff8cc4603a6f618828c867d1897febf4878bf65], Ref[refs/tags/adam-parent-0.15.0=84d5eff081a23f0a89a7ac404bcbd323ded85fbc], Ref[refs/tags/adam-parent-0.16.0=2c13fe359fcf625e15239a8524d6a7e816c403ea], Ref[refs/tags/adam-parent-0.6.0=cc1d98f4cb95b63f7f4b2ab4768843cce51a3a9c], Ref[refs/tags/adam-parent-0.7.0=ae98bda53403e6e6beba563ed0f2ae0098c4c667], Ref[refs/tags/adam-parent-0.7.1=2cf7bd30494b4e3404f483678b1256abd64c448f], Ref[refs/tags/adam-parent-0.7.2=ea4b962ac2f702cddcc1503def5f454f1f2a1899], Ref[refs/tags/adam-parent-0.8.0=8d645f2f93be3147bec0700f80e3175aeea58265], Ref[refs/tags/adam-parent-0.9.0=4b07e5859e447f24a72b4b3143c1f09e9880409c]] ]
[info] Resolved tag [ adam-parent-0.10.0 ] [ PersonIdent[Matt Massie, massie@cs.berkeley.edu, Tue May 13 10:10:10 2014 -0700] ], points at [ commit da00095d62f2747362bf91911f68fc5408f0e038 0 ------ ]
[info] Resolved tag [ adam-parent-0.11.0 ] [ PersonIdent[Matt Massie, massie@cs.berkeley.edu, Mon Jun 2 12:09:53 2014 -0700] ], points at [ commit 69d0041748b3ca27a66f133808789baa1908c795 0 ------ ]
[info] Resolved tag [ adam-parent-0.12.0 ] [ PersonIdent[Matt Massie, massie@cs.berkeley.edu, Tue Jun 17 15:26:30 2014 -0700] ], points at [ commit 6a0afefc41eb51f456c5019f3ba0cff751719cbf 0 ------ ]
[info] Resolved tag [ adam-parent-0.12.1 ] [ PersonIdent[Frank Austin Nothaft, fnothaft@alumni.stanford.edu, Thu Jul 24 06:02:27 2014 -0700] ], points at [ commit 3149912aaee759a2028999fb6b208a7bb0f28e36 0 ------ ]
[info] Resolved tag [ adam-parent-0.13.0 ] [ PersonIdent[Frank Austin Nothaft, fnothaft@alumni.stanford.edu, Wed Aug 20 09:36:51 2014 -0700] ], points at [ commit 211dadbf26bf6ec85eb749c006241d93a47e874a 0 ------ ]
[info] Resolved tag [ adam-parent-0.14.0 ] [ PersonIdent[Matt Massie, massie@cs.berkeley.edu, Tue Sep 16 16:16:41 2014 -0700] ], points at [ commit b02ce94c60398869ce988f034f6893679621b014 0 ------ ]
[info] Resolved tag [ adam-parent-0.15.0 ] [ PersonIdent[Matt Massie, massie@cs.berkeley.edu, Wed Nov 26 08:21:49 2014 -0800] ], points at [ commit 52bcce96a5597e73fc73f5b6e4b24e2d274f547d 0 ------ ]
[info] Resolved tag [ adam-parent-0.16.0 ] [ PersonIdent[Matt Massie, massie@cs.berkeley.edu, Wed Feb 18 13:23:30 2015 -0800] ], points at [ commit f78e2e9eb634459abd4ca7532b453c05ec50dd94 0 ------ ]
[info] Resolved tag [ adam-parent-0.6.0 ] [ PersonIdent[Matt Massie, massie@cs.berkeley.edu, Sun Jan 19 12:43:54 2014 -0800] ], points at [ commit 5fe4f1cdca401bdfb036c3ce4132259abc56d019 0 ------ ]
[info] Resolved tag [ adam-parent-0.7.0 ] [ PersonIdent[Matt Massie, massie@cs.berkeley.edu, Thu Mar 13 13:39:45 2014 -0700] ], points at [ commit 7779e20ea95fcac63b2f52d81854073706485145 0 ------ ]
[info] Resolved tag [ adam-parent-0.7.1 ] [ PersonIdent[Matt Massie, massie@cs.berkeley.edu, Sat Mar 15 10:33:14 2014 -0700] ], points at [ commit e5d1805abccc198cc62270e9acb507ec10c699ae 0 ------ ]
[info] Resolved tag [ adam-parent-0.7.2 ] [ PersonIdent[Matt Massie, massie@cs.berkeley.edu, Thu Mar 27 17:22:48 2014 -0700] ], points at [ commit b1f1fdf61b693afb58cba7d724124fdc29deaf47 0 ------ ]
[info] Resolved tag [ adam-parent-0.8.0 ] [ PersonIdent[Matt Massie, massie@cs.berkeley.edu, Fri Apr 4 08:44:57 2014 -0700] ], points at [ commit 62dcdec9f5f87845c4135a524efecb55b3a528c7 0 ------ ]
[info] Resolved tag [ adam-parent-0.9.0 ] [ PersonIdent[Matt Massie, massie@cs.berkeley.edu, Wed Apr 9 15:09:12 2014 -0700] ], points at [ commit 00923efeacdfcc30397b335d73523418fc6099bd 0 ------ ]
[info] key [ commit 3149912aaee759a2028999fb6b208a7bb0f28e36 0 ------ ], tags => [ [DatedRevTag{id=c75c4dc7b49bd66b0953e2413b7df6802df13c32, tagName='adam-parent-0.12.1', date=July 24, 2014 6:02:27 AM PDT}] ]
[info] key [ commit 69d0041748b3ca27a66f133808789baa1908c795 0 ------ ], tags => [ [DatedRevTag{id=0699a48a62f72eaae134dbc7e99cf68400ef46fd, tagName='adam-parent-0.11.0', date=June 2, 2014 12:09:53 PM PDT}] ]
[info] key [ commit e5d1805abccc198cc62270e9acb507ec10c699ae 0 ------ ], tags => [ [DatedRevTag{id=2cf7bd30494b4e3404f483678b1256abd64c448f, tagName='adam-parent-0.7.1', date=March 15, 2014 10:33:14 AM PDT}] ]
[info] key [ commit 62dcdec9f5f87845c4135a524efecb55b3a528c7 0 ------ ], tags => [ [DatedRevTag{id=8d645f2f93be3147bec0700f80e3175aeea58265, tagName='adam-parent-0.8.0', date=April 4, 2014 8:44:57 AM PDT}] ]
[info] key [ commit b02ce94c60398869ce988f034f6893679621b014 0 ------ ], tags => [ [DatedRevTag{id=5ff8cc4603a6f618828c867d1897febf4878bf65, tagName='adam-parent-0.14.0', date=September 16, 2014 4:16:41 PM PDT}] ]
[info] key [ commit da00095d62f2747362bf91911f68fc5408f0e038 0 ------ ], tags => [ [DatedRevTag{id=313086f04601dc782692d060e6f708dd09ff0a18, tagName='adam-parent-0.10.0', date=May 13, 2014 10:10:10 AM PDT}] ]
[info] key [ commit 00923efeacdfcc30397b335d73523418fc6099bd 0 ------ ], tags => [ [DatedRevTag{id=4b07e5859e447f24a72b4b3143c1f09e9880409c, tagName='adam-parent-0.9.0', date=April 9, 2014 3:09:12 PM PDT}] ]
[info] key [ commit b1f1fdf61b693afb58cba7d724124fdc29deaf47 0 ------ ], tags => [ [DatedRevTag{id=ea4b962ac2f702cddcc1503def5f454f1f2a1899, tagName='adam-parent-0.7.2', date=March 27, 2014 5:22:48 PM PDT}] ]
[info] key [ commit 52bcce96a5597e73fc73f5b6e4b24e2d274f547d 0 ------ ], tags => [ [DatedRevTag{id=84d5eff081a23f0a89a7ac404bcbd323ded85fbc, tagName='adam-parent-0.15.0', date=November 26, 2014 8:21:49 AM PST}] ]
[info] key [ commit 5fe4f1cdca401bdfb036c3ce4132259abc56d019 0 ------ ], tags => [ [DatedRevTag{id=cc1d98f4cb95b63f7f4b2ab4768843cce51a3a9c, tagName='adam-parent-0.6.0', date=January 19, 2014 12:43:54 PM PST}] ]
[info] key [ commit 6a0afefc41eb51f456c5019f3ba0cff751719cbf 0 ------ ], tags => [ [DatedRevTag{id=d923656d2ed75c86530308314a55c5640f9ea3e8, tagName='adam-parent-0.12.0', date=June 17, 2014 3:26:30 PM PDT}] ]
[info] key [ commit f78e2e9eb634459abd4ca7532b453c05ec50dd94 0 ------ ], tags => [ [DatedRevTag{id=2c13fe359fcf625e15239a8524d6a7e816c403ea, tagName='adam-parent-0.16.0', date=February 18, 2015 1:23:30 PM PST}] ]
[info] key [ commit 7779e20ea95fcac63b2f52d81854073706485145 0 ------ ], tags => [ [DatedRevTag{id=ae98bda53403e6e6beba563ed0f2ae0098c4c667, tagName='adam-parent-0.7.0', date=March 13, 2014 1:39:45 PM PDT}] ]
[info] key [ commit 211dadbf26bf6ec85eb749c006241d93a47e874a 0 ------ ], tags => [ [DatedRevTag{id=50557f60e03ebaaa07316578eca6864f5522ac18, tagName='adam-parent-0.13.0', date=August 20, 2014 9:36:51 AM PDT}] ]
[info] Created map: [ {commit 69d0041748b3ca27a66f133808789baa1908c795 0 ------=[adam-parent-0.11.0], commit 62dcdec9f5f87845c4135a524efecb55b3a528c7 0 ------=[adam-parent-0.8.0], commit 00923efeacdfcc30397b335d73523418fc6099bd 0 ------=[adam-parent-0.9.0], commit 52bcce96a5597e73fc73f5b6e4b24e2d274f547d 0 ------=[adam-parent-0.15.0], commit 5fe4f1cdca401bdfb036c3ce4132259abc56d019 0 ------=[adam-parent-0.6.0], commit 211dadbf26bf6ec85eb749c006241d93a47e874a 0 ------=[adam-parent-0.13.0], commit 3149912aaee759a2028999fb6b208a7bb0f28e36 0 ------=[adam-parent-0.12.1], commit e5d1805abccc198cc62270e9acb507ec10c699ae 0 ------=[adam-parent-0.7.1], commit b02ce94c60398869ce988f034f6893679621b014 0 ------=[adam-parent-0.14.0], commit da00095d62f2747362bf91911f68fc5408f0e038 0 ------=[adam-parent-0.10.0], commit b1f1fdf61b693afb58cba7d724124fdc29deaf47 0 ------=[adam-parent-0.7.2], commit 6a0afefc41eb51f456c5019f3ba0cff751719cbf 0 ------=[adam-parent-0.12.0], commit f78e2e9eb634459abd4ca7532b453c05ec50dd94 0 ------=[adam-parent-0.16.0], commit 7779e20ea95fcac63b2f52d81854073706485145 0 ------=[adam-parent-0.7.0]} ]
[info] HEAD is [ 07c17ba195cb0e24ece8ebe3f618e321a65e5350 ]
[info] Repo is in dirty state [ true ]
[info] git.commit.id.describe adam-parent-0.16.0-18-g07c17ba
[info] git.commit.id 07c17ba195cb0e24ece8ebe3f618e321a65e5350
[info] git.commit.id.abbrev 07c17ba
[info] git.commit.user.name Matt Massie
[info] git.commit.user.email massie@cs.berkeley.edu
[info] git.commit.message.full Merge pull request #613 from fnothaft/contig-kmer-counting

[ADAM-612] Point to proper k-mer counters.
[info] git.commit.message.short Merge pull request #613 from fnothaft/contig-kmer-counting
[info] git.commit.time 04.03.2015 @ 22:28:34 PST
[info] git.remote.origin.url https://github.com/bigdatagenomics/adam.git
shouldExclude.apply(git.commit.user.email) = true
shouldExclude.apply(git.commit.user.name) = true
shouldExclude.apply(git.build.user.name) = true
shouldExclude.apply(git.build.user.email) = true
[info] git.build.time 15.03.2015 @ 16:51:05 PDT
[info] found property git.commit.id.abbrev
[info] found property git.commit.message.full
[info] found property git.commit.id
[info] found property git.commit.message.short
[info] found property git.commit.id.describe
[info] found property git.branch
[info] found property git.commit.time
[info] found property git.build.time
[info] found property git.remote.origin.url
[info] Writing properties file to [ /Users/mlyons/Workbench/git/spark/adam/adam-core/src/main/resources/git.properties ] (for module  ADAM: Core1 )...
[info] ADAM ] project ADAM
[info] ADAM: Core ] project ADAM: Core
[info] ADAM: APIs for Java ] project ADAM: APIs for Java
[info] ADAM: CLI ] project ADAM: CLI
[INFO]
[INFO] --- build-helper-maven-plugin:1.7:add-source (add-source) @ adam-core ---
[INFO] Source directory: /Users/mlyons/Workbench/git/spark/adam/adam-core/src/main/scala added.
[INFO]
[INFO] --- scalariform-maven-plugin:0.1.4:format (default-cli) @ adam-core ---
[INFO] Modified 0 of 169 .scala files
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ adam-core ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources
[INFO]
[INFO] --- scala-maven-plugin:3.1.5:compile (scala-compile-first) @ adam-core ---
[WARNING]  Expected all dependencies to require Scala version: 2.10.4
[WARNING]  org.bdgenomics.bdg-utils:bdg-utils-misc:0.1.1 requires scala version: 2.10.4
[WARNING]  org.bdgenomics.bdg-utils:bdg-utils-misc:0.1.1 requires scala version: 2.10.4
[WARNING]  org.bdgenomics.bdg-utils:bdg-utils-metrics:0.1.1 requires scala version: 2.10.4
[WARNING]  org.bdgenomics.bdg-utils:bdg-utils-parquet:0.1.1 requires scala version: 2.10.4
[WARNING]  org.scoverage:scalac-scoverage-plugin_2.10:0.99.2 requires scala version: 2.10.4
[WARNING]  org.bdgenomics.adam:adam-core:0.16.1-SNAPSHOT requires scala version: 2.10.4
[WARNING]  com.twitter:chill_2.10:0.5.0 requires scala version: 2.10.4
[WARNING]  org.spark-project.akka:akka-remote_2.10:2.3.4-spark requires scala version: 2.10.4
[WARNING]  org.spark-project.akka:akka-actor_2.10:2.3.4-spark requires scala version: 2.10.4
[WARNING]  org.spark-project.akka:akka-slf4j_2.10:2.3.4-spark requires scala version: 2.10.4
[WARNING]  org.apache.spark:spark-core_2.10:1.2.0 requires scala version: 2.10.4
[WARNING]  org.json4s:json4s-jackson_2.10:3.2.10 requires scala version: 2.10.4
[WARNING]  org.json4s:json4s-core_2.10:3.2.10 requires scala version: 2.10.4
[WARNING]  org.json4s:json4s-ast_2.10:3.2.10 requires scala version: 2.10.4
[WARNING]  org.json4s:json4s-core_2.10:3.2.10 requires scala version: 2.10.0
[WARNING] Multiple versions of scala libraries detected!
[INFO] Using zinc server for incremental compilation
[info] Compiling 114 Scala sources and 2 Java sources to /Users/mlyons/Workbench/git/spark/adam/adam-core/target/scala-2.10.4/classes...
[warn] /Users/mlyons/Workbench/git/spark/adam/adam-core/src/main/scala/org/bdgenomics/adam/rdd/Coverage.scala:137: match may not be exhaustive.
[warn] It would fail on the following input: (None, None)
[warn]     val points: RDD[OrientedPoint] = pairs.flatMap {
[warn]                                                    ^
[warn] there were 2 deprecation warning(s); re-run with -deprecation for details
[warn] there were 31 feature warning(s); re-run with -feature for details
[warn] three warnings found
[error] warning: [options] bootstrap class path not set in conjunction with -source 1.7
[error]
[error]
[error] An annotation processor threw an uncaught exception.
[error] Consult the following stack trace for details.
[error] java.lang.NoClassDefFoundError: com/sun/tools/javac/main/OptionName
[error] 	at com.google.java.contract.core.apt.AnnotationProcessor.setupPaths(AnnotationProcessor.java:225)
[error] 	at com.google.java.contract.core.apt.AnnotationProcessor.init(AnnotationProcessor.java:148)
[error] 	at com.sun.tools.javac.processing.JavacProcessingEnvironment$ProcessorState.<init>(JavacProcessingEnvironment.java:500)
[error] 	at com.sun.tools.javac.processing.JavacProcessingEnvironment$DiscoveredProcessors$ProcessorStateIterator.next(JavacProcessingEnvironment.java:597)
[error] 	at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:690)
[error] 	at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)
[error] 	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)
[error] 	at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)
[error] 	at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)
[error] 	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)
[error] 	at com.sun.tools.javac.main.Main.compile(Main.java:523)
[error] 	at com.sun.tools.javac.main.Main.compile(Main.java:381)
[error] 	at com.sun.tools.javac.main.Main.compile(Main.java:370)
[error] 	at com.sun.tools.javac.main.Main.compile(Main.java:361)
[error] 	at com.sun.tools.javac.Main.compile(Main.java:56)
[error] 	at com.sun.tools.javac.Main.main(Main.java:42)
[error] Caused by: java.lang.ClassNotFoundException: com.sun.tools.javac.main.OptionName
[error] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:372)
[error] 	at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
[error] 	at java.security.AccessController.doPrivileged(Native Method)
[error] 	at java.net.URLClassLoader.findClass(URLClassLoader.java:360)
[error] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
[error] 	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
[error] 	... 16 more
[error] Compile failed at Mar 15, 2015 4:51:16 PM [7.121s]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] ADAM ............................................... SUCCESS [  4.261 s]
[INFO] ADAM: Core ......................................... FAILURE [ 11.997 s]
[INFO] ADAM: APIs for Java ................................ SKIPPED
[INFO] ADAM: CLI .......................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 16.431 s
[INFO] Finished at: 2015-03-15T16:51:16-07:00
[INFO] Final Memory: 24M/303M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:3.1.5:compile (scala-compile-first) on project adam-core: Execution scala-compile-first of goal net.alchim31.maven:scala-maven-plugin:3.1.5:compile failed. CompileFailed -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
[ERROR]
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :adam-core```

Maven info:
```----------------[15-03-15]-[16:52]--------------------
~/Workbench/git/spark/adam(branch:master*) » mvn --version                                                                                                                         
Apache Maven 3.2.3 (33f8c3e1027c3ddde99d3cdebad2656a31e8fdf4; 2014-08-11T13:58:10-07:00)
Maven home: /usr/local/Cellar/maven/3.2.3/libexec
Java version: 1.7.0_75, vendor: Oracle Corporation
Java home: /Library/Java/JavaVirtualMachines/jdk1.7.0_75.jdk/Contents/Home/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "10.9.5", arch: "x86_64", family: "mac"```

I thought the problem was resolved but I still have 
```
ERROR] An annotation processor threw an uncaught exception.
[ERROR] Consult the following stack trace for details.
[ERROR] java.lang.NoClassDefFoundError: com/sun/tools/javac/main/OptionName
[ERROR] 	at com.google.java.contract.core.apt.AnnotationProcessor.setupPaths(AnnotationProcessor.java:225)
[ERROR] 	at com.google.java.contract.core.apt.AnnotationProcessor.init(AnnotationProcessor.java:148)
```
when I try to build latest snapshot

That's weird, I thought that the OptionName error only happened when
compiling with Java 8; in fact I thought that it is the failure mode when
attempting to compile with Java 8
<https://github.com/bigdatagenomics/adam/issues/198>. It seems like
something must be off with your workspace, though I don't know what it
could be from looking at the output you pasted.

On Sun, Mar 15, 2015 at 7:57 PM Mike Lyons <notifications@github.com> wrote:

> I'm having a hard time compiling with the latest java 7. Can't compile
> with 6, as it's not supported, and can't with 8 for obvious reasons.
> ```~/Workbench/git/spark/adam(branch:master*) » mvn clean package
> -DskipTests
>
> [INFO] Scanning for projects...
> [INFO]
> ------------------------------------------------------------------------
> [INFO] Reactor Build Order:
> [INFO]
> [INFO] ADAM
> [INFO] ADAM: Core
> [INFO] ADAM: APIs for Java
> [INFO] ADAM: CLI
> [INFO]
> [INFO]
> ------------------------------------------------------------------------
> [INFO] Building ADAM 0.16.1-SNAPSHOT
> [INFO]
> ------------------------------------------------------------------------
> [INFO]
> [INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ adam-parent ---
> [INFO] Deleting /Users/mlyons/Workbench/git/spark/adam/target
> [INFO]
> [INFO] --- maven-enforcer-plugin:1.0:enforce (enforce-versions) @
> adam-parent ---
> [INFO]
> [INFO] --- maven-enforcer-plugin:1.0:enforce (enforce-maven) @ adam-parent
> ---
> [INFO]
> [INFO] --- scalariform-maven-plugin:0.1.4:format (default-cli) @
> adam-parent ---
> [INFO] Modified 0 of 205 .scala files
> [INFO]
> [INFO] --- scala-maven-plugin:3.1.5:compile (scala-compile-first) @
> adam-parent ---
> [INFO] No sources to compile
> [INFO]
> [INFO] --- scala-maven-plugin:3.1.5:testCompile (scala-test-compile-first)
> @ adam-parent ---
> [INFO] No sources to compile
> [INFO]
> [INFO]
> ------------------------------------------------------------------------
> [INFO] Building ADAM: Core 0.16.1-SNAPSHOT
> [INFO]
> ------------------------------------------------------------------------
> [INFO]
> [INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ adam-core ---
> [INFO] Deleting /Users/mlyons/Workbench/git/spark/adam/adam-core/target
> [INFO]
> [INFO] --- maven-enforcer-plugin:1.0:enforce (enforce-versions) @
> adam-core ---
> [INFO]
> [INFO] --- maven-enforcer-plugin:1.0:enforce (enforce-maven) @ adam-core
> ---
> [INFO]
> [INFO] --- git-commit-id-plugin:2.1.9:revision (default) @ adam-core ---
> [info] dotGitDirectory /Users/mlyons/Workbench/git/spark/adam/.git
> [info] git.build.user.name mike_lyons
> [info] git.build.user.email mike.lyons@invitae.com
> [info] git.branch master
> [info] Tag refs [
> [Ref[refs/tags/adam-parent-0.10.0=313086f04601dc782692d060e6f708dd09ff0a18],
> Ref[refs/tags/adam-parent-0.11.0=0699a48a62f72eaae134dbc7e99cf68400ef46fd],
> Ref[refs/tags/adam-parent-0.12.0=d923656d2ed75c86530308314a55c5640f9ea3e8],
> Ref[refs/tags/adam-parent-0.12.1=c75c4dc7b49bd66b0953e2413b7df6802df13c32],
> Ref[refs/tags/adam-parent-0.13.0=50557f60e03ebaaa07316578eca6864f5522ac18],
> Ref[refs/tags/adam-parent-0.14.0=5ff8cc4603a6f618828c867d1897febf4878bf65],
> Ref[refs/tags/adam-parent-0.15.0=84d5eff081a23f0a89a7ac404bcbd323ded85fbc],
> Ref[refs/tags/adam-parent-0.16.0=2c13fe359fcf625e15239a8524d6a7e816c403ea],
> Ref[refs/tags/adam-parent-0.6.0=cc1d98f4cb95b63f7f4b2ab4768843cce51a3a9c],
> Ref[refs/tags/adam-parent-0.7.0=ae98bda53403e6e6beba563ed0f2ae0098c4c667],
> Ref[refs/tags/adam-parent-0.7.1=2cf7bd30494b4e3404f483678b1256abd64c448f],
> Ref[refs/tags/adam-parent-0.7.2=ea4b962ac2f702cddcc1503def5f454f1f2a1899],
> Ref[refs/tags/adam-parent-0.8.0=8d645f2f93be3147bec0700f80e3175aeea58265],
> Ref[refs/tags/adam-parent-0.9.0=4b07e5859e447f24a72b4b3143c1f09e9880409c]] ]
> [info] Resolved tag [ adam-parent-0.10.0 ] [ PersonIdent[Matt Massie,
> massie@cs.berkeley.edu, Tue May 13 10:10:10 2014 -0700] ], points at [
> commit da00095
> <https://github.com/bigdatagenomics/adam/commit/da00095d62f2747362bf91911f68fc5408f0e038>
> 0 ------ ]
> [info] Resolved tag [ adam-parent-0.11.0 ] [ PersonIdent[Matt Massie,
> massie@cs.berkeley.edu, Mon Jun 2 12:09:53 2014 -0700] ], points at [
> commit 69d0041
> <https://github.com/bigdatagenomics/adam/commit/69d0041748b3ca27a66f133808789baa1908c795>
> 0 ------ ]
> [info] Resolved tag [ adam-parent-0.12.0 ] [ PersonIdent[Matt Massie,
> massie@cs.berkeley.edu, Tue Jun 17 15:26:30 2014 -0700] ], points at [
> commit 6a0afef
> <https://github.com/bigdatagenomics/adam/commit/6a0afefc41eb51f456c5019f3ba0cff751719cbf>
> 0 ------ ]
> [info] Resolved tag [ adam-parent-0.12.1 ] [ PersonIdent[Frank Austin
> Nothaft, fnothaft@alumni.stanford.edu, Thu Jul 24 06:02:27 2014 -0700] ],
> points at [ commit 3149912
> <https://github.com/bigdatagenomics/adam/commit/3149912aaee759a2028999fb6b208a7bb0f28e36>
> 0 ------ ]
> [info] Resolved tag [ adam-parent-0.13.0 ] [ PersonIdent[Frank Austin
> Nothaft, fnothaft@alumni.stanford.edu, Wed Aug 20 09:36:51 2014 -0700] ],
> points at [ commit 211dadb
> <https://github.com/bigdatagenomics/adam/commit/211dadbf26bf6ec85eb749c006241d93a47e874a>
> 0 ------ ]
> [info] Resolved tag [ adam-parent-0.14.0 ] [ PersonIdent[Matt Massie,
> massie@cs.berkeley.edu, Tue Sep 16 16:16:41 2014 -0700] ], points at [
> commit b02ce94
> <https://github.com/bigdatagenomics/adam/commit/b02ce94c60398869ce988f034f6893679621b014>
> 0 ------ ]
> [info] Resolved tag [ adam-parent-0.15.0 ] [ PersonIdent[Matt Massie,
> massie@cs.berkeley.edu, Wed Nov 26 08:21:49 2014 -0800] ], points at [
> commit 52bcce9
> <https://github.com/bigdatagenomics/adam/commit/52bcce96a5597e73fc73f5b6e4b24e2d274f547d>
> 0 ------ ]
> [info] Resolved tag [ adam-parent-0.16.0 ] [ PersonIdent[Matt Massie,
> massie@cs.berkeley.edu, Wed Feb 18 13:23:30 2015 -0800] ], points at [
> commit f78e2e9
> <https://github.com/bigdatagenomics/adam/commit/f78e2e9eb634459abd4ca7532b453c05ec50dd94>
> 0 ------ ]
> [info] Resolved tag [ adam-parent-0.6.0 ] [ PersonIdent[Matt Massie,
> massie@cs.berkeley.edu, Sun Jan 19 12:43:54 2014 -0800] ], points at [
> commit 5fe4f1c
> <https://github.com/bigdatagenomics/adam/commit/5fe4f1cdca401bdfb036c3ce4132259abc56d019>
> 0 ------ ]
> [info] Resolved tag [ adam-parent-0.7.0 ] [ PersonIdent[Matt Massie,
> massie@cs.berkeley.edu, Thu Mar 13 13:39:45 2014 -0700] ], points at [
> commit 7779e20
> <https://github.com/bigdatagenomics/adam/commit/7779e20ea95fcac63b2f52d81854073706485145>
> 0 ------ ]
> [info] Resolved tag [ adam-parent-0.7.1 ] [ PersonIdent[Matt Massie,
> massie@cs.berkeley.edu, Sat Mar 15 10:33:14 2014 -0700] ], points at [
> commit e5d1805
> <https://github.com/bigdatagenomics/adam/commit/e5d1805abccc198cc62270e9acb507ec10c699ae>
> 0 ------ ]
> [info] Resolved tag [ adam-parent-0.7.2 ] [ PersonIdent[Matt Massie,
> massie@cs.berkeley.edu, Thu Mar 27 17:22:48 2014 -0700] ], points at [
> commit b1f1fdf
> <https://github.com/bigdatagenomics/adam/commit/b1f1fdf61b693afb58cba7d724124fdc29deaf47>
> 0 ------ ]
> [info] Resolved tag [ adam-parent-0.8.0 ] [ PersonIdent[Matt Massie,
> massie@cs.berkeley.edu, Fri Apr 4 08:44:57 2014 -0700] ], points at [
> commit 62dcdec
> <https://github.com/bigdatagenomics/adam/commit/62dcdec9f5f87845c4135a524efecb55b3a528c7>
> 0 ------ ]
> [info] Resolved tag [ adam-parent-0.9.0 ] [ PersonIdent[Matt Massie,
> massie@cs.berkeley.edu, Wed Apr 9 15:09:12 2014 -0700] ], points at [
> commit 00923ef
> <https://github.com/bigdatagenomics/adam/commit/00923efeacdfcc30397b335d73523418fc6099bd>
> 0 ------ ]
> [info] key [ commit 3149912
> <https://github.com/bigdatagenomics/adam/commit/3149912aaee759a2028999fb6b208a7bb0f28e36>
> 0 ------ ], tags => [
> [DatedRevTag{id=c75c4dc7b49bd66b0953e2413b7df6802df13c32,
> tagName='adam-parent-0.12.1', date=July 24, 2014 6:02:27 AM PDT}] ]
> [info] key [ commit 69d0041
> <https://github.com/bigdatagenomics/adam/commit/69d0041748b3ca27a66f133808789baa1908c795>
> 0 ------ ], tags => [
> [DatedRevTag{id=0699a48a62f72eaae134dbc7e99cf68400ef46fd,
> tagName='adam-parent-0.11.0', date=June 2, 2014 12:09:53 PM PDT}] ]
> [info] key [ commit e5d1805
> <https://github.com/bigdatagenomics/adam/commit/e5d1805abccc198cc62270e9acb507ec10c699ae>
> 0 ------ ], tags => [
> [DatedRevTag{id=2cf7bd30494b4e3404f483678b1256abd64c448f,
> tagName='adam-parent-0.7.1', date=March 15, 2014 10:33:14 AM PDT}] ]
> [info] key [ commit 62dcdec
> <https://github.com/bigdatagenomics/adam/commit/62dcdec9f5f87845c4135a524efecb55b3a528c7>
> 0 ------ ], tags => [
> [DatedRevTag{id=8d645f2f93be3147bec0700f80e3175aeea58265,
> tagName='adam-parent-0.8.0', date=April 4, 2014 8:44:57 AM PDT}] ]
> [info] key [ commit b02ce94
> <https://github.com/bigdatagenomics/adam/commit/b02ce94c60398869ce988f034f6893679621b014>
> 0 ------ ], tags => [
> [DatedRevTag{id=5ff8cc4603a6f618828c867d1897febf4878bf65,
> tagName='adam-parent-0.14.0', date=September 16, 2014 4:16:41 PM PDT}] ]
> [info] key [ commit da00095
> <https://github.com/bigdatagenomics/adam/commit/da00095d62f2747362bf91911f68fc5408f0e038>
> 0 ------ ], tags => [
> [DatedRevTag{id=313086f04601dc782692d060e6f708dd09ff0a18,
> tagName='adam-parent-0.10.0', date=May 13, 2014 10:10:10 AM PDT}] ]
> [info] key [ commit 00923ef
> <https://github.com/bigdatagenomics/adam/commit/00923efeacdfcc30397b335d73523418fc6099bd>
> 0 ------ ], tags => [
> [DatedRevTag{id=4b07e5859e447f24a72b4b3143c1f09e9880409c,
> tagName='adam-parent-0.9.0', date=April 9, 2014 3:09:12 PM PDT}] ]
> [info] key [ commit b1f1fdf
> <https://github.com/bigdatagenomics/adam/commit/b1f1fdf61b693afb58cba7d724124fdc29deaf47>
> 0 ------ ], tags => [
> [DatedRevTag{id=ea4b962ac2f702cddcc1503def5f454f1f2a1899,
> tagName='adam-parent-0.7.2', date=March 27, 2014 5:22:48 PM PDT}] ]
> [info] key [ commit 52bcce9
> <https://github.com/bigdatagenomics/adam/commit/52bcce96a5597e73fc73f5b6e4b24e2d274f547d>
> 0 ------ ], tags => [
> [DatedRevTag{id=84d5eff081a23f0a89a7ac404bcbd323ded85fbc,
> tagName='adam-parent-0.15.0', date=November 26, 2014 8:21:49 AM PST}] ]
> [info] key [ commit 5fe4f1c
> <https://github.com/bigdatagenomics/adam/commit/5fe4f1cdca401bdfb036c3ce4132259abc56d019>
> 0 ------ ], tags => [
> [DatedRevTag{id=cc1d98f4cb95b63f7f4b2ab4768843cce51a3a9c,
> tagName='adam-parent-0.6.0', date=January 19, 2014 12:43:54 PM PST}] ]
> [info] key [ commit 6a0afef
> <https://github.com/bigdatagenomics/adam/commit/6a0afefc41eb51f456c5019f3ba0cff751719cbf>
> 0 ------ ], tags => [
> [DatedRevTag{id=d923656d2ed75c86530308314a55c5640f9ea3e8,
> tagName='adam-parent-0.12.0', date=June 17, 2014 3:26:30 PM PDT}] ]
> [info] key [ commit f78e2e9
> <https://github.com/bigdatagenomics/adam/commit/f78e2e9eb634459abd4ca7532b453c05ec50dd94>
> 0 ------ ], tags => [
> [DatedRevTag{id=2c13fe359fcf625e15239a8524d6a7e816c403ea,
> tagName='adam-parent-0.16.0', date=February 18, 2015 1:23:30 PM PST}] ]
> [info] key [ commit 7779e20
> <https://github.com/bigdatagenomics/adam/commit/7779e20ea95fcac63b2f52d81854073706485145>
> 0 ------ ], tags => [
> [DatedRevTag{id=ae98bda53403e6e6beba563ed0f2ae0098c4c667,
> tagName='adam-parent-0.7.0', date=March 13, 2014 1:39:45 PM PDT}] ]
> [info] key [ commit 211dadb
> <https://github.com/bigdatagenomics/adam/commit/211dadbf26bf6ec85eb749c006241d93a47e874a>
> 0 ------ ], tags => [
> [DatedRevTag{id=50557f60e03ebaaa07316578eca6864f5522ac18,
> tagName='adam-parent-0.13.0', date=August 20, 2014 9:36:51 AM PDT}] ]
> [info] Created map: [ {commit 69d0041
> <https://github.com/bigdatagenomics/adam/commit/69d0041748b3ca27a66f133808789baa1908c795>
> 0 ------=[adam-parent-0.11.0], commit 62dcdec
> <https://github.com/bigdatagenomics/adam/commit/62dcdec9f5f87845c4135a524efecb55b3a528c7>
> 0 ------=[adam-parent-0.8.0], commit 00923ef
> <https://github.com/bigdatagenomics/adam/commit/00923efeacdfcc30397b335d73523418fc6099bd>
> 0 ------=[adam-parent-0.9.0], commit 52bcce9
> <https://github.com/bigdatagenomics/adam/commit/52bcce96a5597e73fc73f5b6e4b24e2d274f547d>
> 0 ------=[adam-parent-0.15.0], commit 5fe4f1c
> <https://github.com/bigdatagenomics/adam/commit/5fe4f1cdca401bdfb036c3ce4132259abc56d019>
> 0 ------=[adam-parent-0.6.0], commit 211dadb
> <https://github.com/bigdatagenomics/adam/commit/211dadbf26bf6ec85eb749c006241d93a47e874a>
> 0 ------=[adam-parent-0.13.0], commit 3149912
> <https://github.com/bigdatagenomics/adam/commit/3149912aaee759a2028999fb6b208a7bb0f28e36>
> 0 ------=[adam-parent-0.12.1], commit e5d1805
> <https://github.com/bigdatagenomics/adam/commit/e5d1805abccc198cc62270e9acb507ec10c699ae>
> 0 ------=[adam-parent-0.7.1], commit b02ce94
> <https://github.com/bigdatagenomics/adam/commit/b02ce94c60398869ce988f034f6893679621b014>
> 0 ------=[adam-parent-0.14.0], commit da00095
> <https://github.com/bigdatagenomics/adam/commit/da00095d62f2747362bf91911f68fc5408f0e038>
> 0 ------=[adam-parent-0.10.0], commit b1f1fdf
> <https://github.com/bigdatagenomics/adam/commit/b1f1fdf61b693afb58cba7d724124fdc29deaf47>
> 0 ------=[adam-parent-0.7.2], commit 6a0afef
> <https://github.com/bigdatagenomics/adam/commit/6a0afefc41eb51f456c5019f3ba0cff751719cbf>
> 0 ------=[adam-parent-0.12.0], commit f78e2e9
> <https://github.com/bigdatagenomics/adam/commit/f78e2e9eb634459abd4ca7532b453c05ec50dd94>
> 0 ------=[adam-parent-0.16.0], commit 7779e20
> <https://github.com/bigdatagenomics/adam/commit/7779e20ea95fcac63b2f52d81854073706485145>
> 0 ------=[adam-parent-0.7.0]} ]
> [info] HEAD is [ 07c17ba
> <https://github.com/bigdatagenomics/adam/commit/07c17ba195cb0e24ece8ebe3f618e321a65e5350>
> ]
> [info] Repo is in dirty state [ true ]
> [info] git.commit.id.describe adam-parent-0.16.0-18-g07c17ba
> [info] git.commit.id 07c17ba
> <https://github.com/bigdatagenomics/adam/commit/07c17ba195cb0e24ece8ebe3f618e321a65e5350>
> [info] git.commit.id.abbrev 07c17ba
> <https://github.com/bigdatagenomics/adam/commit/07c17ba195cb0e24ece8ebe3f618e321a65e5350>
> [info] git.commit.user.name Matt Massie
> [info] git.commit.user.email massie@cs.berkeley.edu
> [info] git.commit.message.full Merge pull request #613
> <https://github.com/bigdatagenomics/adam/pull/613> from
> fnothaft/contig-kmer-counting
>
> [ADAM-612] Point to proper k-mer counters.
> [info] git.commit.message.short Merge pull request #613
> <https://github.com/bigdatagenomics/adam/pull/613> from
> fnothaft/contig-kmer-counting
> [info] git.commit.time 04.03.2015 @ 22:28:34 PST
> [info] git.remote.origin.url https://github.com/bigdatagenomics/adam.git
> shouldExclude.apply(git.commit.user.email) = true
> shouldExclude.apply(git.commit.user.name) = true
> shouldExclude.apply(git.build.user.name) = true
> shouldExclude.apply(git.build.user.email) = true
> [info] git.build.time 15.03.2015 @ 16:51:05 PDT
> [info] found property git.commit.id.abbrev
> [info] found property git.commit.message.full
> [info] found property git.commit.id
> [info] found property git.commit.message.short
> [info] found property git.commit.id.describe
> [info] found property git.branch
> [info] found property git.commit.time
> [info] found property git.build.time
> [info] found property git.remote.origin.url
> [info] Writing properties file to
> /Users/mlyons/Workbench/git/spark/adam/adam-core/src/main/resources/git.properties
> ...
> [info] ADAM ] project ADAM
> [info] ADAM: Core ] project ADAM: Core
> [info] ADAM: APIs for Java ] project ADAM: APIs for Java
> [info] ADAM: CLI ] project ADAM: CLI
> [INFO]
> [INFO] --- build-helper-maven-plugin:1.7:add-source (add-source) @
> adam-core ---
> [INFO] Source directory:
> /Users/mlyons/Workbench/git/spark/adam/adam-core/src/main/scala added.
> [INFO]
> [INFO] --- scalariform-maven-plugin:0.1.4:format (default-cli) @ adam-core
> ---
> [INFO] Modified 0 of 169 .scala files
> [INFO]
> [INFO] --- maven-resources-plugin:2.6:resources (default-resources) @
> adam-core ---
> [INFO] Using 'UTF-8' encoding to copy filtered resources.
> [INFO] Copying 2 resources
> [INFO]
> [INFO] --- scala-maven-plugin:3.1.5:compile (scala-compile-first) @
> adam-core ---
> [WARNING] Expected all dependencies to require Scala version: 2.10.4
> [WARNING] org.bdgenomics.bdg-utils:bdg-utils-misc:0.1.1 requires scala
> version: 2.10.4
> [WARNING] org.bdgenomics.bdg-utils:bdg-utils-misc:0.1.1 requires scala
> version: 2.10.4
> [WARNING] org.bdgenomics.bdg-utils:bdg-utils-metrics:0.1.1 requires scala
> version: 2.10.4
> [WARNING] org.bdgenomics.bdg-utils:bdg-utils-parquet:0.1.1 requires scala
> version: 2.10.4
> [WARNING] org.scoverage:scalac-scoverage-plugin_2.10:0.99.2 requires scala
> version: 2.10.4
> [WARNING] org.bdgenomics.adam:adam-core:0.16.1-SNAPSHOT requires scala
> version: 2.10.4
> [WARNING] com.twitter:chill_2.10:0.5.0 requires scala version: 2.10.4
> [WARNING] org.spark-project.akka:akka-remote_2.10:2.3.4-spark requires
> scala version: 2.10.4
> [WARNING] org.spark-project.akka:akka-actor_2.10:2.3.4-spark requires
> scala version: 2.10.4
> [WARNING] org.spark-project.akka:akka-slf4j_2.10:2.3.4-spark requires
> scala version: 2.10.4
> [WARNING] org.apache.spark:spark-core_2.10:1.2.0 requires scala version:
> 2.10.4
> [WARNING] org.json4s:json4s-jackson_2.10:3.2.10 requires scala version:
> 2.10.4
> [WARNING] org.json4s:json4s-core_2.10:3.2.10 requires scala version: 2.10.4
> [WARNING] org.json4s:json4s-ast_2.10:3.2.10 requires scala version: 2.10.4
> [WARNING] org.json4s:json4s-core_2.10:3.2.10 requires scala version: 2.10.0
> [WARNING] Multiple versions of scala libraries detected!
> [INFO] Using zinc server for incremental compilation
> [info] Compiling 114 Scala sources and 2 Java sources to
> /Users/mlyons/Workbench/git/spark/adam/adam-core/target/scala-2.10.4/classes...
> [warn]
> /Users/mlyons/Workbench/git/spark/adam/adam-core/src/main/scala/org/bdgenomics/adam/rdd/Coverage.scala:137:
> match may not be exhaustive.
> [warn] It would fail on the following input: (None, None)
> [warn] val points: RDD[OrientedPoint] = pairs.flatMap {
> [warn] ^
> [warn] there were 2 deprecation warning(s); re-run with -deprecation for
> details
> [warn] there were 31 feature warning(s); re-run with -feature for details
> [warn] three warnings found
> [error] warning: [options] bootstrap class path not set in conjunction
> with -source 1.7
> [error]
> [error]
> [error] An annotation processor threw an uncaught exception.
> [error] Consult the following stack trace for details.
> [error] java.lang.NoClassDefFoundError: com/sun/tools/javac/main/OptionName
> [error] at
> com.google.java.contract.core.apt.AnnotationProcessor.setupPaths(AnnotationProcessor.java:225)
> [error] at
> com.google.java.contract.core.apt.AnnotationProcessor.init(AnnotationProcessor.java:148)
> [error] at
> com.sun.tools.javac.processing.JavacProcessingEnvironment$ProcessorState.(JavacProcessingEnvironment.java:500)
> [error] at
> com.sun.tools.javac.processing.JavacProcessingEnvironment$DiscoveredProcessors$ProcessorStateIterator.next(JavacProcessingEnvironment.java:597)
> [error] at
> com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:690)
> [error] at
> com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)
> [error] at
> com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)
> [error] at
> com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)
> [error] at
> com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)
> [error] at
> com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)
> [error] at com.sun.tools.javac.main.Main.compile(Main.java:523)
> [error] at com.sun.tools.javac.main.Main.compile(Main.java:381)
> [error] at com.sun.tools.javac.main.Main.compile(Main.java:370)
> [error] at com.sun.tools.javac.main.Main.compile(Main.java:361)
> [error] at com.sun.tools.javac.Main.compile(Main.java:56)
> [error] at com.sun.tools.javac.Main.main(Main.java:42)
> [error] Caused by: java.lang.ClassNotFoundException:
> com.sun.tools.javac.main.OptionName
> [error] at java.net.URLClassLoader$1.run(URLClassLoader.java:372)
> [error] at java.net.URLClassLoader$1.run(URLClassLoader.java:361)
> [error] at java.security.AccessController.doPrivileged(Native Method)
> [error] at java.net.URLClassLoader.findClass(URLClassLoader.java:360)
> [error] at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
> [error] at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
> [error] ... 16 more
> [error] Compile failed at Mar 15, 2015 4:51:16 PM [7.121s]
> [INFO]
> ------------------------------------------------------------------------
> [INFO] Reactor Summary:
> [INFO]
> [INFO] ADAM ............................................... SUCCESS [
> 4.261 s]
> [INFO] ADAM: Core ......................................... FAILURE [
> 11.997 s]
> [INFO] ADAM: APIs for Java ................................ SKIPPED
> [INFO] ADAM: CLI .......................................... SKIPPED
> [INFO]
> ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO]
> ------------------------------------------------------------------------
> [INFO] Total time: 16.431 s
> [INFO] Finished at: 2015-03-15T16:51:16-07:00
> [INFO] Final Memory: 24M/303M
> [INFO]
> ------------------------------------------------------------------------
> [ERROR] Failed to execute goal
> net.alchim31.maven:scala-maven-plugin:3.1.5:compile (scala-compile-first)
> on project adam-core: Execution scala-compile-first of goal
> net.alchim31.maven:scala-maven-plugin:3.1.5:compile failed. CompileFailed
> -> [Help 1]
> [ERROR]
> [ERROR] To see the full stack trace of the errors, re-run Maven with the
> -e switch.
> [ERROR] Re-run Maven using the -X switch to enable full debug logging.
> [ERROR]
> [ERROR] For more information about the errors and possible solutions,
> please read the following articles:
> [ERROR] [Help 1]
> http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
> [ERROR]
> [ERROR] After correcting the problems, you can resume the build with the
> command
> [ERROR] mvn -rf :adam-core```
>
> Maven info:
>
> ~/Workbench/git/spark/adam(branch:master*) » mvn --version
> Apache Maven 3.2.3 (33f8c3e1027c3ddde99d3cdebad2656a31e8fdf4; 2014-08-11T13:58:10-07:00)
> Maven home: /usr/local/Cellar/maven/3.2.3/libexec
> Java version: 1.7.0_75, vendor: Oracle Corporation
> Java home: /Library/Java/JavaVirtualMachines/jdk1.7.0_75.jdk/Contents/Home/jre
> Default locale: en_US, platform encoding: UTF-8
> OS name: "mac os x", version: "10.9.5", arch: "x86_64", family: "mac"```
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bigdatagenomics/adam/issues/540#issuecomment-81305226>
> .
>

--- a/adam-core/pom.xml
+++ b/adam-core/pom.xml
@@ -178,11 +178,7 @@
       <artifactId>hadoop-bam</artifactId>
     </dependency>
     <dependency>
-      <groupId>org.seqdoop</groupId>
-      <artifactId>cofoja</artifactId>
-    </dependency>
-    <dependency>
-      <groupId>org.seqdoop</groupId>
+      <groupId>com.github.samtools</groupId>
       <artifactId>htsjdk</artifactId>
     </dependency>
     <dependency>

--- a/pom.xml
+++ b/pom.xml
@@ -27,6 +27,7 @@
     <hadoop.version>2.2.0</hadoop.version>
     <scoverage.version>0.99.2</scoverage.version>
     <utils.version>0.1.1</utils.version>
+    <htsjdk.version>1.129</htsjdk.version>
   </properties>
   
   <modules>
@@ -432,16 +433,17 @@
         <groupId>org.seqdoop</groupId>
         <artifactId>hadoop-bam</artifactId>
         <version>7.0.0</version>
+        <exclusions>
+          <exclusion>
+            <groupId>com.seqdoop</groupId>
+            <artifactId>htsjdk</artifactId>
+          </exclusion>
+        </exclusions>
       </dependency>
       <dependency>
-        <groupId>org.seqdoop</groupId>
-        <artifactId>cofoja</artifactId>
-        <version>1.1-r150</version>
-      </dependency>
-      <dependency>
-        <groupId>org.seqdoop</groupId>
+        <groupId>com.github.samtools</groupId>
         <artifactId>htsjdk</artifactId>
-        <version>1.118</version>
+        <version>${htsjdk.version}</version>
       </dependency>
       <dependency>
         <groupId>args4j</groupId>

